### PR TITLE
 Add primaries for the ctime, mtime, and atime attributes

### PR DESCRIPTION
**Pull request by ekinanp**
Contributions to this project require sign-off consistent with the [Developers Certificate of Origin](https://developercertificate.org). This can be as simple as using `git commit -s` on each commit.

### CODE REVIEW

File: cmd/find/grammar.go [hunk 1]
```diff
@@ -13,104 +13,98 @@ type atom struct {
...
-	parensOp,
-	namePrimary,
-	truePrimary,
-	falsePrimary,
-}
-
-// Map of <token> => <atom>. This is initialized in ParsePredicate. Unfortunately,
-// we cannot intialize atoms here because doing so would result in an initialization
-// loop compiler error (e.g. notOp => atoms => notOp).
+// Map of <token> => <atom>.
```
Comment (MikaelSmith) [inline cmd/find/grammar.go]: I think a comment that this is populated by calls to `newAtom` would be helpful.

File: cmd/find/grammar.go [hunk 2]
```diff
@@ -13,104 +13,98 @@ type atom struct {
...
 	precedence int
 	combine    func(p1 Predicate, p2 Predicate) Predicate
 }
 
-var allBinaryOps = []*binaryOp{
-	andOp,
-	orOp,
-}
-
 // Map of <token> => <binaryOp>
```
Comment (MikaelSmith) [inline cmd/find/grammar.go]: Same comment here for newBinaryOp.

File: cmd/find/timeAttrPrimary.go [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+	}
+	return duration, roundDiff
+}
+
+// timeAttrPrimary => -<name> (+|-)?((\d+)|(\d+[smhdw])
+//
+// where s => seconds, m => minutes, h => hours, d => days, w => weeks
+//
+// Example inputs:
+//   -mtime 1      (true if the difference between the entry's mtime and startTime is exactly 1 24-hour period)
```
Comment (MikaelSmith) [inline cmd/find/timeAttrPrimary.go]: Is there any rounding on that? Alternatively, is there any way to express "between 23 and 24 hours"? I guess that could be `-mtime +23h -and -mtime -24h`.

File: cmd/find/timeAttrPrimary.go [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+		}
+	}
+	return duration, roundDiff
+}
+
+// timeAttrPrimary => -<name> (+|-)?((\d+)|(\d+[smhdw])
+//
+// where s => seconds, m => minutes, h => hours, d => days, w => weeks
+//
+// Example inputs:
```
Comment (MikaelSmith) [inline cmd/find/timeAttrPrimary.go]: How can we document all these primaries? Starting to need a man-page :)

File: cmd/find/timeAttrPrimary.go (hunk 3 above)
Comment (ekinanp) [reply cmd/find/timeAttrPrimary.go]: Yeah, that seems to be the only way to do it. We could consider adding alternate syntax for these primaries that supports something like that? Wouldn't be hard to do if people find `and`'ing the `mtimes` annoying.

File: cmd/find/timeAttrPrimary.go (hunk 4 above)
Comment (ekinanp) [reply cmd/find/timeAttrPrimary.go]: Either a really long usage, or a man page. Could go with a long usage for now, then the man page?

I'd like to handle the usage separately though, at least after the `Size` primary's been added (and maybe after the full traversal logic [i.e. `depth`])

File: cmd/find/timeAttrPrimary.go (hunk 3 above)
Comment (MikaelSmith) [reply cmd/find/timeAttrPrimary.go]: I think the way I outlined is fine. Not sure exact numbers will be that useful in practice.

File: cmd/find/timeAttrPrimary.go (hunk 4 above)
Comment (MikaelSmith) [reply cmd/find/timeAttrPrimary.go]: Sure.

**Pull request merged**